### PR TITLE
introduce reload_zsh

### PR DESCRIPTION
**Pull request by holymonson**
## Standards checklist:

- [x] The PR title is descriptive.
- [x] The PR doesn't replicate another PR which is already open.
- [x] I have read the contribution guide and followed all the instructions.
- [x] The code follows the code style guide detailed in the wiki.
- [x] The code is mine or it's from somewhere with an MIT-compatible license.
- [x] The code is efficient, to the best of my ability, and does not waste computer resources.
- [x] The code is stable and I have tested it myself, to the best of my abilities.

## Changes:

Introduce `reload_zsh()`, which could easily remove the zcompdump file and reload zsh.

## Other comments:

This should be associated with #8802 and #8801, and better with #9074.
Close #8800.

### CODE REVIEW

**Comment by bartekpacia** [comment]
After investigating the topic this looks like a good idea. Thank you.

File: plugins/zsh_reload/README.md [hunk 1]
```diff
@@ -1,23 +1,3 @@
 # zsh_reload plugin
 
-The zsh_reload plugin defines a function to reload the zsh session with
```
Comment (Fhiltyphil88) [inline plugins/zsh_reload/README.md:3]:
```suggestion
The zsh_reload plugin defines a function to reload the zsh session with
```

**Review by Fhiltyphil88 — COMMENTED** [review]
@atmos

**Review by TheTechRobo — APPROVED** [review]

**Comment by mcornella** [comment]
I'd prefer this be in the `omz` CLI so that if this gets documented somewhere external to OMZ at least it is made very clear that it comes from Oh My Zsh and therefore needs it. You don't need to do anything, I'll quickly make the edits myself.

**Pull request closed without merging**